src/_gettsim/transfers/kinderzuschl/kost_unterk.py

```diff
@@ -21,71 +20,131 @@ def kinderzuschl_kost_unterk_m_bg(
     -------
 
     """
-    warmmiete_m_hh = bruttokaltmiete_m_hh + heizkosten_m_hh
-    anteil_warmmiete_m_bg = warmmiete_m_hh * _anteil_personen_in_haushalt_bg
 
-    out = _kinderzuschl_wohnbedarf_eltern_anteil_bg * anteil_warmmiete_m_bg
+    return (
+        bruttokaltmiete_m_bg + heizkosten_m_bg
+    ) * _kinderzuschl_wohnbedarf_eltern_anteil_bg
 
-    return out
 
-
-def bruttokaltmiete_m_bg(
+def bruttokaltmiete_m_bg(  # noqa: PLR0913
+    anz_paare_hh: int,
     bruttokaltmiete_m_hh: float,
-    _anteil_personen_in_haushalt_bg: float,
+    anz_kinder_bis_17_bg: int,
+    anz_kinder_bis_17_hh: int,
+    anz_erwachsene_bg: int,
+    anz_erwachsene_hh: int,
+    kinderzuschl_params: dict,
 ) -> float:
-    """Share of household's monthly rent attributed to the Bedarfsgemeinschaft.
+    """Monthly rent attributed to the Bedarfsgemeinschaft.
+
+    The rent is split among a household's Bedarfsgemeinschaften according to their needs
+    specified in the Existenzminimumsbericht.
 
     Parameters
     ----------
+    anz_paare_hh
+        See :func:`anz_paare_hh`.
     bruttokaltmiete_m_hh
         See basic input variable :ref:`bruttokaltmiete_m_hh <bruttokaltmiete_m_hh>`.
-    _anteil_personen_in_haushalt_bg
-        See :func:`_anteil_personen_in_haushalt_bg`.
+    anz_kinder_bis_17_bg
+        See :func:`anz_kinder_bis_17_bg`.
+    anz_kinder_bis_17_hh
+        See :func:`anz_kinder_bis_17_hh`.
+    anz_erwachsene_bg
+        See :func:`anz_erwachsene_bg`.
+    anz_erwachsene_hh
+        See :func:`anz_erwachsene_hh`.
+    kinderzuschl_params
+        See params documentation :ref:`kinderzuschl_params <kinderzuschl_params>`.
 
     Returns
     -------
 
     """
-    return bruttokaltmiete_m_hh * _anteil_personen_in_haushalt_bg
+    bedarf_hh = (
+        anz_paare_hh
+        * kinderzuschl_params["existenzminimum"]["kosten_der_unterkunft"]["paare"]
+        + (anz_erwachsene_hh - anz_paare_hh * 2)
+        * kinderzuschl_params["existenzminimum"]["kosten_der_unterkunft"]["single"]
+        + anz_kinder_bis_17_hh
+        * kinderzuschl_params["existenzminimum"]["kosten_der_unterkunft"]["kinder"]
+    )
 
+    if anz_erwachsene_bg == 1:
+        bedarf_bg = (
+            anz_erwachsene_bg
+            * kinderzuschl_params["existenzminimum"]["kosten_der_unterkunft"]["single"]
+            + anz_kinder_bis_17_bg
+            * kinderzuschl_params["existenzminimum"]["kosten_der_unterkunft"]["kinder"]
+        )
+    else:
+        bedarf_bg = (
+            kinderzuschl_params["existenzminimum"]["kosten_der_unterkunft"]["paare"]
+            + anz_kinder_bis_17_bg
+            * kinderzuschl_params["existenzminimum"]["kosten_der_unterkunft"]["kinder"]
+        )
 
-def heizkosten_m_bg(
+    return bruttokaltmiete_m_hh * (bedarf_bg / bedarf_hh)
+
+
+def heizkosten_m_bg(  # noqa: PLR0913
```

Comment:
For `kinderzuschlag`, we use the same factor for bruttokaltmiete and heizkosten which is based on the sum of `ex_min["kosten_der_unterkunft"]` and `ex_min["heizkosten"]`. It shouldn't matter much, but we should have an issue on that to check what is the correct way.